opentreemap/treemap/lib/page_of_items.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.core.urlresolvers import reverse
+
+
+class UrlParams(object):
+    def __init__(self, url_name, *url_args, **params):
+        self._url = reverse(url_name, args=url_args) + '?'
+        self._params = params
+
+    def params(self, *keys):
+        return self._param_string(keys, self._params)
+
+    def url(self, *keys, **overrides):
+        params = self._params
+        if overrides:
+            keys += tuple(overrides.keys())
+            params = dict(params.items() + overrides.items())
+        return self._url + self._param_string(keys, params)
+
+    @staticmethod
+    def _param_string(keys, params):
+        return '&'.join(['%s=%s' % (key, params[key]) for key in keys])
+
+
+def make_filter_context(urlizer, filter_value, option_data):
```

Comment:
I agree that your suggestion would simplify the code somewhat. But currently `UrlParams` is a fairly generic class and I hesitate to add such specific view-context details.